NuKeeper/Engine/RepositoryFilter.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Threading.Tasks;
+using NuKeeper.Configuration;
+using NuKeeper.Github;
+using NuKeeper.Inspection.Logging;
+using Octokit;
+
+namespace NuKeeper.Engine
+{
+    public class RepositoryFilter : IRepositoryFilter
+    {
+        private readonly IGithub _githubClient;
+        private readonly INuKeeperLogger _logger;
+
+        public RepositoryFilter(IGithub githubClient, INuKeeperLogger logger)
+        {
+            _githubClient = githubClient;
+            _logger = logger;
+        }
+
+        public async Task<bool> ContainsDotNetProjects(RepositorySettings repository)
+        {
+            var request = new SearchCodeRequest("\"packages.config\" OR \".csproj\" OR \".fsproj\" OR \".vbproj\"")
+            {
+                Repos = new RepositoryCollection {{repository.RepositoryOwner, repository.RepositoryName}},
+                In = new []{CodeInQualifier.Path},
+                PerPage = 1
+            };
+            try
+            {
+                var result = await _githubClient.Search(request);
+                if (result.TotalCount <= 0)
+                {
+                    _logger.Verbose(
+                        $"Repository {repository.RepositoryOwner}/{repository.RepositoryName} contains no .NET code on the default branch, skipping.");
+                    return false;
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.Error("GitHub Repository search failed.", ex);
+            }
+
+            return true;
```

Comment:
Is it correct to return `true` in the error case? I suppose it is, since you can't rule out .NET code.

Reply:
That's the idea, yes - we will then inspect the local clone of the repo.